File: zohodesk-portal-core/ios/RNZohoDeskPortalCore/RNZohoDeskPortalHome.m
//
//  RNZDPortalCore.m
//  RNZDPortalCore
//
//  Created by Rajesh R on 16/09/24.
//

#import "RNZohoDeskPortalHome.h"
#import <React/RCTLog.h>
@import ZohoDeskPortalCore;
@import ZohoDeskPortalConfiguration;

@implementation RNZohoDeskPortalHome

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}
RCT_EXPORT_MODULE()

RCT_EXPORT_METHOD(show){
  [ZDPortalHome showWithTitle:nil navigationMode:ZDPNavigationModePresent];
}



RCT_EXPORT_METHOD(updateConfiguration:(NSDictionary *)configDictionary){
  ZDPHomeConfiguration *configuration = [[ZDPHomeConfiguration alloc] init];
  
  configuration.enableCommunity = [configDictionary[@"enableCommunity"] boolValue];
  configuration.enableHelpCenter = [configDictionary[@"enableHelpCenter"] boolValue];
  configuration.enableMyTicket = [configDictionary[@"enableMyTicket"] boolValue];
  configuration.enableCreateTicket = [configDictionary[@"enableCreateTicket"] boolValue];
  configuration.enableAddTopic = [configDictionary[@"enableAddTopic"] boolValue];
  configuration.showChat = [configDictionary[@"showChat"] boolValue];
  configuration.showGC = [configDictionary[@"showGC"] boolValue];
  configuration.showAnswerBot = [configDictionary[@"showAnswerBot"] boolValue];
  configuration.showBM = [configDictionary[@"showBM"] boolValue];
  
  [ZDPortalHome updateConfigurationWith:configuration];
}

@end

File: zohodesk-portal-core/ios/RNZohoDeskPortalCore/RNZohoDeskPortalHome.swift
import Foundation
import ZohoDeskPortalCore
import ZohoDeskPortalConfiguration
import React

@objc(RNZohoDeskPortalHome)
class RNZohoDeskPortalHome: NSObject {

  @objc
  func methodQueue() -> DispatchQueue {
    return DispatchQueue.main
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func show() {
    ZDPortalHome.show(withTitle: nil, navigationMode: .present)
  }

  @objc
  func updateConfiguration(_ configDictionary: [String: Any]) {
    let configuration = ZDPHomeConfiguration()

    configuration.enableCommunity = configDictionary["enableCommunity"] as? Bool ?? true
    configuration.enableHelpCenter = configDictionary["enableHelpCenter"] as? Bool ?? true
    configuration.enableMyTicket = configDictionary["enableMyTicket"] as? Bool ?? true
    configuration.enableCreateTicket = configDictionary["enableCreateTicket"] as? Bool ?? true
    configuration.enableAddTopic = configDictionary["enableAddTopic"] as? Bool ?? true
    configuration.showChat = configDictionary["showChat"] as? Bool ?? true
    configuration.showGC = configDictionary["showGC"] as? Bool ?? true
    configuration.showAnswerBot = configDictionary["showAnswerBot"] as? Bool ?? true
    configuration.showBM = configDictionary["showBM"] as? Bool ?? true
    
    ZDPortalHome.updateConfiguration(with: configuration)
  }
}

